core/src/main/java/hudson/PluginManager.java

```diff
@@ -1955,6 +1955,8 @@ public HttpResponse doUploadPlugin(StaplerRequest req) throws IOException, Servl
                 }
             } catch (MalformedURLException e) {
                 return FormValidation.error(e.getMessage());
+            } catch (URISyntaxException e) {
+                throw new RuntimeException(e);
```

Comment:
why are we blowing up here and not disoplaying the error to the user?

Reply:
I have replaced
```
        } catch (URISyntaxException e) {
            throw new RuntimeException(e);
```
with
```
        } catch (URISyntaxException e) {
            throw new URISyntaxException(e.getReason(), e.getInput());
```